core/src/main/java/com/linecorp/armeria/server/PathMappings.java

```diff
@@ -118,16 +116,34 @@ public PathMappings(int cacheSize) {
         for (int i = 0; i < size; i++) {
             final Entry<PathMapping, T> e = patterns.get(i);
             final PathMapping mapping = e.getKey();
-            final PathMappingResult mappingResult = mapping.apply(path, query);
+            final PathMappingResult mappingResult = mapping.apply(mappingCtx);
             if (mappingResult.isPresent()) {
-                result = PathMapped.of(mapping, mappingResult, e.getValue());
-                break;
+                // If the result is not the best matching, we keep it and try to find the better matching again.
+                if (mappingResult.isHighestScore()) {
+                    result = PathMapped.of(mapping, mappingResult, e.getValue());
```

Comment:
Seems like the comment is not right. Isn't the result the best match and thus we should stop looking for more?